extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,235 @@
+            Type _ when targetType == typeof(BlobContainerClient) => CreateBlobContainerClient(connectionString, containerName),
+            _ => await DeserializeToTargetObject(targetType, connectionString, containerName, blobName)
+        };
+
+        private IEnumerable<object> ToTargetCollectionType(Type targetType, IEnumerable<object> array) => targetType switch
+        {
+            Type _ when targetType == typeof(BlobClient) => array.Select(p => (BlobClient)p),
+            Type _ when targetType == typeof(String) => array.Select(p => GetBlobContentString((BlobClient)p).GetAwaiter().GetResult()),
+            _ => throw new Exception()
+        };
+
+        private async Task<object?> DeserializeToTargetObject(Type targetType, string connectionString, string containerName, string blobName)
+        {
+            var content = await GetBlobString(connectionString, containerName, blobName);
+            return JsonConvert.DeserializeObject(content, targetType);
+        }
+
+        private async Task<string> GetBlobString(string connectionString, string containerName, string blobName, ConverterContext context = null)
+        {
+            var client = CreateBlobReference<BlobClient>(connectionString, containerName, blobName, context);
+            return await GetBlobContentString(client);
+        }
+
+        private async Task<string> GetBlobContentString(BlobClient client)
+        {
+            var download = await client.DownloadContentAsync();
+            return download.Value.Content.ToString();
+        }
+
+        private async Task<Byte[]> GetBlobBinaryData(string connectionString, string containerName, string blobName)
+        {
+            using (MemoryStream stream = new MemoryStream())
+            {
+                var client = CreateBlobReference<BlobClient>(connectionString, containerName, blobName);
+                await client.DownloadToAsync(stream);
+                return stream.ToArray();
+            }
+        }
+
+        private async Task<Stream> GetBlobStream(string connectionString, string containerName, string blobName)
+        {
+            var client = CreateBlobReference<BlobClient>(connectionString, containerName, blobName);
+            var download = await client.DownloadStreamingAsync();
+            return download.Value.Content;
+        }
+
+        private BlobContainerClient CreateBlobContainerClient(string connectionString, string containerName)
+        {
+            BlobContainerClient container = new(connectionString, containerName);
+            return container;
+        }
+
+        private T CreateBlobReference<T>(string connectionString, string containerName, string blobName, ConverterContext context = null) where T : BlobBaseClient
+        {
+            if (string.IsNullOrEmpty(blobName))
+            {
+                throw new ArgumentNullException(nameof(blobName));
```

Comment:
Is any of the public methods in the caller chain handling this exception being thrown (catching and returning a ConversionStatus.Failed()?

Reply:
Good question, will need to test this out and confirm

Reply:
Looking up the stack, I don't see anything handling the exception. Do you suggest we return a failure status here? We could turn this into `TryCreateBlobClient` and return a bool and have the bubble up to ConvertAsync to return Failure result?

Reply:
Added a catch!